src/Microsoft.OData.Client/Metadata/ClientTypeUtil.cs

```diff
@@ -28,6 +29,8 @@ internal static class ClientTypeUtil
         /// <summary>A static empty PropertyInfo array.</summary>
         internal static readonly PropertyInfo[] EmptyPropertyInfoArray = new PropertyInfo[0];
 
+        internal static ConcurrentDictionary<Type, ODataTypeInfo> ODataTypeInfoCache { get; set; } = new ConcurrentDictionary<Type, ODataTypeInfo>();

```

Comment:
1) one ConcurrentDictionary here
2) two ConcurrentDictionarys in each ODataTypeInfo?
It seems too much?